staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -1361,7 +1361,8 @@ func buildOpenAPIModelsForApply(staticOpenAPISpec *spec.Swagger, crd *apiextensi
 	specs := []*spec.Swagger{}
 	for _, v := range crd.Spec.Versions {
 		// Defaults are not pruned here, but before being served.
-		s, err := builder.BuildSwagger(crd, v.Name, builder.Options{V2: false, StripValueValidation: true, StripNullable: true, AllowNonStructural: false})
+		// See flag description in builder.go for flag usage
+		s, err := builder.BuildOpenAPIV2(crd, v.Name, builder.Options{V2: true, OverrideToStructuralOpenAPIV2: true, StripValueValidation: true, StripNullable: true, AllowNonStructural: false})
```

Comment:
To be fair, I tried to look at it and I couldn't find a risky way to improve without refactoring a bunch. I'd hope we get rid of OpenAPI v2 rather than spend too much time refactoring v2/v3. WDYT?

Reply:
I'd like to completely get rid of the V2 field in the options in a follow up PR and refactor the code so the options make more sense. I'm not sure how well we can improve this section in this PR without going through the refactor.

Reply:
Yeah, I'd love that, but I agree that it might be too ambitious for now.